code/modules/tgui_panel/tgui_panel.dm

```diff
@@ -37,6 +37,9 @@
  * Initializes tgui panel.
  */
 /datum/tgui_panel/proc/initialize(force = FALSE)
+	set waitfor = FALSE
+	// Minimal sleep to defer initialization to after client constructor
```

Comment:
Why is this not explicitly done in the client constructor? This seems hacky.

Reply:
We do not want to delay client constructor with any sleeps, and it's totally fine if initialize yields control back. I did this because

1. I thought this would improve loading times for the client and somehow fix reconnect issues of chat loading bug
2. Goonchat was doing it too.

Reply:
I don't mean the sleep to be done in the client constructor, I mean just intiailizing the tgui_panel at the end. I must be misunderstanding something.

Reply:
It would still block the constructor for an unreasonably long time.